File: HackerNewsReaderJava/app/src/main/java/com/example/hackernewsreader/WebViewActivity.java
package com.example.hackernewsreader;

import android.content.Intent;
import android.database.Cursor;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public class WebViewActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web_view);

        Intent intent = getIntent();
        int newsIndex = intent.getIntExtra("newsId", 1);

        WebView webView = findViewById(R.id.webView);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new WebViewClient());

        Cursor c;
        c = MainActivity.sqLiteDatabase.rawQuery("SELECT * FROM news WHERE id = '"+newsIndex+"'", null);
        c.moveToFirst();

        String webViewDestination = c.getString(c.getColumnIndex("url"));
        webView.loadUrl(webViewDestination);
    }
}

File: HackerNewsReader/app/src/main/java/com/example/hackernewsreader/WebViewActivity.kt
package com.example.hackernewsreader

import android.database.Cursor
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.webkit.WebView
import android.webkit.WebViewClient

class WebViewActivity : AppCompatActivity() {
    lateinit var  webView: WebView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web_view)

        val newsIndex = intent.getIntExtra("newsId", 1)

        webView.settings.javaScriptEnabled = true
        webView.webViewClient = WebViewClient()

        val c: Cursor

        c = MainActivity.sqLiteDatabase!!.rawQuery("SELECT * FROM news WHERE id = '"+newsIndex+"'", null)
        c.moveToFirst()

        val webViewDestination = c.getString(c.getColumnIndex("url"))
        webView.loadUrl(webViewDestination)

        c.close()
    }
}